doc/src/modules/combinatorics/fp_groups.rst

```diff
@@ -0,0 +1,272 @@
+  specified positive integer, using the *Low-Index Subgroups* algorithm.
+
+* Algorithms for computing presentations of a subgroup of finite index
+  in a group defined by finite presentation.
+
+For a description of fundamental algorithms of finitely presented groups
+we often make use of *Handbook of Computational Group Theory*.
+
+The Construction of Finitely Presented Groups
+---------------------------------------------
+
+Finitely presented groups are construced by factoring a free group by a
+set of relators. The set of relators is taken in as a list of words in
+generators of free group in SymPy, using a list provides ordering to the
+relators. If the list of relators is empty, the associated free group is
+returned.
+
+Example of construction of a finitely-presented group.
+The symmetric group of degree 4 may be represented as a two generator group
+with presentation `< a, b \mid a^2, b^3, (a*b)^4 >`. Giving the relations as a
+list of relators, the presentation of would be specified as:
+
+>>> F, a, b = free_group("a, b")
+>>> G = FpGroup(F, [a**2, b**3, (a*b)**4])
+>>> G
+<fp group on the generators (a, b)>
+
+Currently groups with relators having presentation like
+`< r, s, t \mid r^2, s^2, t^2, rst = str = trs >` will have to be specified as:
+
+>>> F, r, s, t = free_group("r, s, t")
+>>> G = FpGroup(F, [r**2, s**2, t**2, r*s*t*r**-1*t**-1*s**-1, s*t*r*s**-1*r**-1*t**-1])
+
+obviously this is not a unique way to make such a group, but the point is that
+in case of equality with non-identity the user has to manually do that.
+
+Free Groups and Words
+---------------------
+
+Construction of a Free Group
+````````````````````````````
+
+``free_group("gen0, gen1, \ldots gen_(n-1)")`` constructs a free group ``F`` on ``n``
+generators, where ``n`` is a positive integer.
+The `i`-th generator of `F` may be obtained using the method ``.generators[i]``, `i = 0, \ldots n-1`.
+
+>>> F, x, y = free_group("x, y")
+
+creates a free group of rank 2 and assigns the variables ``x`` and ``y`` to the two
+generators.
+
+>>> F = vfree_group("x, y")
+>>> F
+<free group on the generators (x, y)>
+
+creates a free group ``F`` of rank 2, with tuple of generators ``F.generators``,
+and inserts ``x`` and ``y`` as generators into the global namespace.
+
+>>> F = xfree_group("x, y")
+>>> F
+(<free group on the generators (x, y)>, (x, y))
+
+creates a free groups ``F[0]`` of rank 2, with tuple of generators ``F[1]``.
+
+Construction of words
+`````````````````````
+
+This section is applicable to words of ``FreeGroup`` as well as ``FpGroup``.
+When we say *word* in SymPy, it actually means a `reduced word
+<https://en.wikipedia.org/wiki/Word_(group_theory)#Reduced_words>`_ , since the
+words are automatically reduced. Given a group ``G`` defined on `n` generators
+`x_1, x_2, x_3, \ldots, x_n`, a word is constructed as
+`\prod{x_i^{r_j}}` where `i \in [1, 2, \ldots, n]` and `j \in \mathbb{Z}`.
```

Comment:
There are two ways to denote the interval of integers from 1 to n, either `\{1, 2, \ldots, n\}` (set notation) or `[1, n]` (interval notation, implicitly restricted to integers). In SymPy, I think the set notation should be chosen to avoid confusion.